docs/api_reference/src/quantities.tex

```diff
@@ -109,158 +623,5351 @@
+\tcode{T} otherwise.
+
+\pnum
+\exposid{type-less} meets the requirements of
+the \tcode{Pred} parameter of the expression template algorithms.
+The expression \tcode{\exposidnc{type-less-impl}<T, U>()} evaluates to
+\tcode{true} if \tcode{T} is less than \tcode{U}
+in an implementation-defined total order for types, and
+\tcode{false} otherwise.
+
+\begin{codeblock}
+template<typename... Ts>
+struct @\exposidnc{type-list}@ {};      // \expos
+
+template<typename OneType, typename... Ts>
+struct @\exposidnc{expr-fractions}@ {   // \expos
+  using @\exposidnc{num}@ = @\seebelownc@;  // \expos
+  using @\exposidnc{den}@ = @\seebelownc@;  // \expos
+}
+\end{codeblock}
+
+\pnum
+\exposidnc{expr-fractions} divides an expression template to numerator and denominator parts.
+Unless otherwise specified, let \exposidnc{num} and \exposidnc{den} be \tcode{\exposidnc{type-list}<>}.
+\begin{itemize}
+\item
+If \tcode{sizeof...(Ts) == 1} is \tcode{true}, then \exposidnc{num} is \tcode{\exposidnc{type-list}<Ts...>}.
+\item
+If \tcode{sizeof...(Ts) >= 2} is \tcode{true}, then:
 \begin{itemize}
 \item
-If \tcode{Rep} is a scalar,
-\tcode{\placeholder{Q}} represents the scalar quantity\irefiev{102-02-19}
-of a position vector.
+If the last element of \tcode{Ts} does not match \tcode{per<Us...>},
+then \exposidnc{num} is \tcode{\exposidnc{type-list}<Ts...>}.
 \item
-If \tcode{Rep} is a vector,
-\tcode{\placeholder{Q}} represents a position vector.
-% FIXME What if `Rep` is a tensor?
+Otherwise, \exposidnc{den} is \tcode{\exposidnc{type-list}<Us...>}, and if
+\begin{codeblock}
+sizeof...(Ts) != 2 || !std::is_same_v<Ts...[0], OneType>
+\end{codeblock}
+is \tcode{true}, then
+\exposidnc{num} is \tcode{\exposidnc{type-list}<Ts...[0], Ts...[1], \ldots, Ts...[sizeof...(Ts) - 2]>}.
+\end{itemize}
 \end{itemize}
 
-\rSec1[qty.compat]{Compatibility}
+\begin{itemdecl}
+template<template<typename...> typename To, typename OneType,
+         template<typename, typename> typename Pred = @\exposidnc{type-less}@, typename Lhs, typename Rhs>
+consteval auto @\exposidnc{expr-multiply}@(Lhs, Rhs);        // \expos
 
-\rSec1[qty.one]{Dimension one}
+template<template<typename...> typename To, typename OneType,
+         template<typename, typename> typename Pred = @\exposidnc{type-less}@, typename Lhs, typename Rhs>
+consteval auto @\exposidnc{expr-divide}@(Lhs lhs, Rhs rhs);  // \expos
 
-\rSec1[qty.systems]{Systems}
+template<template<typename...> typename To, typename OneType, typename T>
+consteval auto @\exposidnc{expr-invert}@(T);                 // \expos
 
-\rSec1[qty.chrono]{\tcode{std::chrono} compatibility}
+template<std::intmax_t Num, std::intmax_t Den, template<typename...> typename To,
+         typename OneType, template<typename, typename> typename Pred = @\exposidnc{type-less}@, typename T>
+  requires(Den != 0)
+consteval auto @\exposidnc{expr-pow}@(T);                    // \expos
+\end{itemdecl}
```

Comment:
mp-units has a constrained `type_less_X` for each kind of `X` it passes to these algorithms.
I have replaced them with a single unconstrained _`type-less`_.
Do you forsee any problems with this?
Is this another opportunity to improve the compile times of mp-units?

Reply:
Initially, I used different sorting rules for different entities. Now, all of them just compare type_names. It might be simplified now, but I do not see the reason why it could improve compile times?